src/common/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useState, useId } from "react";
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
+import { CheckIcon, MinusIcon } from "@/common/components/CustomIcon";
+import { CheckboxVariants, checkboxTheme } from "@/common/components/Checkbox";
+
+export type CheckboxProps = CheckboxVariants &
+  CheckboxPrimitive.CheckboxProps & {
+    label: string;
+    indeterminate?: boolean;
+  };
+
+export const Checkbox = ({
+  label,
+  className,
+  size = "md",
+  disabled = false,
+  indeterminate = false,
+  ...props
+}: CheckboxProps) => {
+  const { checked: initialCheckedState, defaultChecked } = props;
+  const initialState = () => {
+    if (initialCheckedState !== undefined) {
+      return initialCheckedState;
+    }
+    if (defaultChecked !== undefined) {
+      return defaultChecked;
+    }
+    return indeterminate ? "indeterminate" : false;
+  };
+  const [checked, setChecked] = useState(initialState());
```

Comment:
it seems like declaring a state is a must in order to use the `"indeterminate"` state

i can't seem to think of a scenario where it would be useful for the consumer to manually define the `checked` state / `onCheckedChange` handler too, is there a scenario where that's useful?

Reply:
Hmm the main point would be how we'll be using the checkbox. E.g how does the consumer know what the state of the checkbox is after the initial state has changed?

Reply:
make sense! updated it at a063e97

i've also added forwardRef (just as previously done in `input`) to allow checkbox to be used in `react-hook-forms` out of the box